### PR TITLE
schema: reduce memory usage when init the schema

### LINKED ISSUE #533: CDC cli hang and OOM
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
When after TiDB cluster running a long time(with much table create and delete). cdc cli create changefeed will hang and then OOM

2. What did you expect to see?
cdc cli create changefeed soon and quick return

3. What did you see instead?
cdc cli eat more then 30G Memory. but ddl history only 1.8G

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        master
        ```

    - TiCDC version (execute `cdc version`):

        ```
        master
        ```

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
reduce memory usage when init the schema, fix #533 

### Release note

- Use TableInfo snapshot to init schema storage, which reduces memory consumption and speeds up the schema storage initialization, especially when DDL history is quite large.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -146,22 +146,19 @@ func (o *Owner) newChangeFeed(
...
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	schemaStorage, err := entry.NewSchemaStorage(jobs, filter)
+	filter, err := util.NewFilter(info.GetConfig())
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	schemaStorage.FlushIneligibleTables(checkpointTs)
```
Comment (amyangfei) [inline cdc/owner.go:163]: func `FlushIneligibleTables` and `flushIneligibleTables` can be removed now

**Comment by leoppro** [comment]
fix #533

**Comment by amyangfei** [comment]
Please resolve conflicts

**Pull request merged**